File: PaseoFrontEnd/src/app/modelos/ronda.ts
import { Estado } from "app/utils/estado";
import { Pedido } from "./pedido";

export class Ronda {

    id: number;
    fechaInicio: Date;
	fechaFin: Date;
	fechaRetiro: Date; 
	rangoHorario: string;
	pedidos: Pedido[];
    estado:Estado;


	constructor(fechaInicio: Date, fechaFin: Date, fechaRetiro: Date, rangoHorario: string, pedidos: Pedido[], estado:Estado) {
        this.fechaInicio=fechaInicio;
        this.fechaFin=fechaFin;
        this.fechaRetiro=fechaRetiro; 
        this.rangoHorario=rangoHorario;
        this.pedidos=pedidos;   
        this.estado=estado;
    }


    public get Pedidos(): Pedido[] {
        return this.pedidos;
    }
    public set Pedidos(value: Pedido[]) {
        this.pedidos = value;
    }

    public get Id(): number {
        return this.id;
    }
    public set Id(value: number) {
        this.id = value;
    }
    public get FechaInicio(): Date {
        return this.fechaInicio;
    }
    public set FechaInicio(value: Date) {
        this.fechaInicio = value;
    }
    public get FechaFin(): Date {
        return this.fechaFin;
    }
    public set FechaFin(value: Date) {
        this.fechaFin = value;
    }
    public get FechaRetiro(): Date {
        return this.fechaRetiro;
    }
    public set FechaRetiro(value: Date) {
        this.fechaRetiro = value;
    }
    public get RangoHorario(): string {
        return this.rangoHorario;
    }
    public set RangoHorario(value: string) {
        this.rangoHorario = value;
    }
}

File: Paseo2/src/main/java/modelos/Ronda.java
package modelos;

import java.util.Date;

import java.util.List;

import javax.persistence.Entity;
import javax.persistence.EnumType;
import javax.persistence.Enumerated;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.OneToMany;

import com.fasterxml.jackson.annotation.JsonIgnore;

import io.swagger.v3.oas.annotations.media.Schema;
import utils.Estado;

@Entity
public class Ronda {
	
    
    @Id @GeneratedValue(strategy=GenerationType.IDENTITY) 
	private Long id;
	private Date fechaInicio;
	private Date fechaFin;
	private Date fechaRetiro; 
	private String rangoHorario;
	
	@Enumerated(EnumType.ORDINAL)
	@Schema(implementation = Estado.class)
    private Estado estado;
	
	@OneToMany(mappedBy = "ronda")
	@JsonIgnore
	private List<Pedido> pedidos;
	
	public Ronda() {
		
	}
	public Ronda(Date fechaInicio, Date fechaFin, Date fechaRetiro, String rangoHorario,Estado estado) {
		super();
		this.fechaInicio = fechaInicio;
		this.fechaFin = fechaFin;
		this.fechaRetiro = fechaRetiro;
		this.rangoHorario = rangoHorario;
		this.estado = estado;
	}
	
	public String getRangoHorario() {
		return rangoHorario;
	}
	public void setRangoHorario(String rangoHorario) {
		this.rangoHorario = rangoHorario;
	}
	public List<Pedido> getPedidos() {
		return pedidos;
	}
	public void setPedidos(List<Pedido> pedidos) {
		this.pedidos = pedidos;
	}
	
	public Long getId() {
		return id;
	}
	
	public Estado getEstado() {
		return estado;
	}
	public void setEstado(Estado estado) {
		this.estado = estado;
	}
	public void setId(Long id) {
		this.id = id;
	}
	
	public Date getFechaInicio() {
		return fechaInicio;
	}
	public void setFechaInicio(Date fechaInicio) {
		this.fechaInicio = fechaInicio;
	}
	public Date getFechaFin() {
		return fechaFin;
	}
	public void setFechaFin(Date fechaFin) {
		this.fechaFin = fechaFin;
	}
	public Date getFechaRetiro() {
		return fechaRetiro;
	}
	public void setFechaRetiro(Date fechaRetiro) {
		this.fechaRetiro = fechaRetiro;
	}
	@Override
	public String toString() {
		return "Ronda [id=" + id + ", fechaInicio=" + fechaInicio + ", fechaFin=" + fechaFin + ", fechaRetiro="
				+ fechaRetiro + ", rangoHorario=" + rangoHorario + ", estado=" + estado + ", pedidos=" + pedidos + "]";
	}
	
	
}
